lib/TPP/Transforms/LinalgTiling.cpp

```diff
@@ -0,0 +1,229 @@
+      if (inductionVars[indexThree][effectiveOffsetThree] == NULL) {
+        inductionVars[indexThree][effectiveOffsetThree] =
+            loopOp.getInductionVar();
+      }
+      innermostForLoop = loopOp;
+      if ((finaltile.size() - 1) == (i + 1)) {
+        Value zeroCst1 = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+        Value ubCst1 = rewriter.create<arith::ConstantIndexOp>(
+            loc, dyn_cast<MemRefType>(linalgOp.getOperand(0).getType())
+                     .getShape()[0]);
+        Value stepCst1 = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+        scf::ForOp redloopOp = rewriter.create<scf::ForOp>(
+            linalgOp.getLoc(), zeroCst1, ubCst1, stepCst1);
+        rewriter.setInsertionPointToStart(redloopOp.getBody());
+        reductionForLoop = redloopOp;
+      }
+    }
+
+    // Creating subviews
+    SmallVector<SmallVector<int64_t>> tiles = {tileShapeM, tileShapeN};
+    for (size_t i = 0; i < linalgOp.getNumOperands(); i++) {
+      SmallVector<int64_t> indices;
+
+      auto input = linalgOp.getOperand(i);
+      auto operandType = input.getType();
+      SmallVector<OpFoldResult> offsets;
+      size_t k = 0;
+      auto tileItr = tileshapes[i].begin();
+      auto tensorShape = dyn_cast<MemRefType>(operandType).getShape();
+      SmallVector<OpFoldResult> shape;
+      SmallVector<OpFoldResult> strides;
+      for (size_t j = 0; j < tensorShape.size(); j++) {
+        if (j < tensorShape.size() - tileSizesIndex[i]) {
+          if (j == ((tensorShape.size() - tileSizesIndex[i]) - 1) &&
+              i < (linalgOp.getNumOperands() - 1)) {
+            offsets.push_back(reductionForLoop.getInductionVar());
+            indices.push_back(tensorShape[j] / 32);
+            shape.push_back(rewriter.getIndexAttr(tensorShape[j] / 32));
+            strides.push_back(rewriter.getIndexAttr(1));
+
+          } else {
+            offsets.push_back(rewriter.getIndexAttr(0));
+            indices.push_back(tensorShape[j]);
+            shape.push_back(rewriter.getIndexAttr(tensorShape[j]));
+            strides.push_back(rewriter.getIndexAttr(1));
+          }
+        } else {
+          shape.push_back(rewriter.getIndexAttr(tensorShape[j] / (*tileItr)));
+          indices.push_back(tensorShape[j] / (*tileItr));
+          strides.push_back(rewriter.getIndexAttr(1));
+          offsets.push_back(
+              inductionVars[i][tensorShape.size() - tileSizesIndex[i] + k]);
+          k++;
+          tileItr++;
+        }
+      }
+
+      auto subview = rewriter.create<memref::SubViewOp>(
+          linalgOp.getLoc(), MemRefType(),
+          input, offsets, shape, strides);
+      linalgOp.setOperand(i, subview);
+    }
+
+    rewriter.setInsertionPoint(innermostForLoop.getBody(),
+                               std::prev(innermostForLoop.getBody()->end(), 1));
+    auto clone = rewriter.clone(*linalgOp);
+    linalgOp.replaceAllUsesWith(clone);
+    if (linalgOp->use_empty())
```

Comment:
After removing all users, why use would be _not_ empty?

Reply:
We are getting `error: operand #0 does not dominate this use`, if we don't call `use_empty()`.

Reply:
Is this for all your tests? If not, can you investigate which cases you get the error and which you don't?

I would be good to know what the IR looks like to see what the problem is, and if correct, add a comment to that line explaining why.